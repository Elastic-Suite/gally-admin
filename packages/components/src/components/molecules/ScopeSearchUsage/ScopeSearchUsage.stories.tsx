import React from 'react'
import { ComponentStory, Meta } from '@storybook/react'
import ScopeSearchUsage from './ScopeSearchUsage'
import { action } from '@storybook/addon-actions'
import catalogMock from '../../../../public/mocks/catalog.json'

export default {
  title: 'Molecules/ScopeSearchUsage',
  component: ScopeSearchUsage,
} as Meta

const Template: ComponentStory<typeof ScopeSearchUsage> = (args) => (
  <ScopeSearchUsage {...args} />
)

export const Default = Template.bind({})

const catalogsOptions = catalogMock['hydra:member'].flatMap((catalog) => {
  return catalog.localizedCatalogs.map((localizedCatalog) => {
    return {
      id: catalog.name,
      value: localizedCatalog['@id'],
      label: localizedCatalog.name,
    }
  })
})

Default.args = {
  catalogsOptions,
  onSubmit: (value: string): void => action('Scope submit')(value),
}
