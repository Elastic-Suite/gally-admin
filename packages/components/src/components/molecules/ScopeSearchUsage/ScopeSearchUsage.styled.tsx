import { styled } from '@mui/system'
import Form from '../../atoms/form/Form'

export const ScopeSearchUsageForm = styled(Form)(({ theme }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'flex-end',
  flexWrap: 'wrap',
  gap: theme.spacing(4),
}))
