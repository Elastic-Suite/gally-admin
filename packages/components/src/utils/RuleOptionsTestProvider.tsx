import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
