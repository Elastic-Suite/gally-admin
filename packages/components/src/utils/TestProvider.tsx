import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
