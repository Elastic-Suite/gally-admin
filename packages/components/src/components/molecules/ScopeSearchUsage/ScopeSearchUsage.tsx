import React, { useState } from 'react'
import { useTranslation } from 'next-i18next'
import { SectionFieldSet } from '../../organisms/CustomForm/CustomForm.styled'
import DoubleDatePicker from '../../atoms/form/DoubleDatePicker'
import DropDown from '../../atoms/form/DropDown'
import { IDoubleDatePickerValues } from '../../atoms/form/DoubleDatePickerWithoutError'
import { IOptions } from '@elastic-suite/gally-admin-shared'
import { ScopeSearchUsageForm } from './ScopeSearchUsage.styled'

interface IScopeSearchUsageProps {
  catalogsOptions: IOptions<string>
  onSubmit?: (value: {
    catalogs: string | string[]
    rangeDate: IDoubleDatePickerValues
  }) => void
}

function ScopeSearchUsage({
  catalogsOptions,
  onSubmit,
}: IScopeSearchUsageProps): JSX.Element {
  const [showErrors, setShowErrors] = useState(false)
  const [catalogValue, setCatalogValue] = useState<string | string[]>()
  const [rangeDateValue, setRangeDateValue] = useState<IDoubleDatePickerValues>(
    {
      fromDate: null,
      toDate: null,
    }
  )
  const { t } = useTranslation('searchUsage')

  function handleSubmit(
    e: React.SyntheticEvent<Element, Event>,
    formIsValid: boolean
  ): void {
    e.preventDefault()
    if (formIsValid && onSubmit) {
      onSubmit({
        catalogs: catalogValue!,
        rangeDate: rangeDateValue,
      })
    }
    setShowErrors(!formIsValid)
  }

  return (
    <SectionFieldSet style={{ padding: 16 }}>
      <ScopeSearchUsageForm
        submitButtonText={t('Apply') || ''}
        onSubmit={handleSubmit}
      >
        <DropDown
          sx={{ minWidth: '300px' }}
          label={t('Scope')}
          options={catalogsOptions}
          placeholder={t('Select a localized catalog') || ''}
          useGroups
          value={catalogValue}
          onChange={setCatalogValue}
          showError={showErrors}
          required
        />

        <DoubleDatePicker
          value={rangeDateValue}
          onChange={setRangeDateValue}
          showError={showErrors}
          required
        />
      </ScopeSearchUsageForm>
    </SectionFieldSet>
  )
}

export default ScopeSearchUsage
