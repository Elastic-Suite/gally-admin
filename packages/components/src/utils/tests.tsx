import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
