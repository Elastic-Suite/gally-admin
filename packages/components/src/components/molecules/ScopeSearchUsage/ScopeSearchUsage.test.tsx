import React from 'react'
import { renderWithProviders } from '../../../utils/tests'
import ScopeSearchUsage from './ScopeSearchUsage'
import catalogMock from '../../../../public/mocks/catalog.json'

describe('ScopeSearchUsage', () => {
  it('match snapshot', () => {
    const catalogsOptions = catalogMock['hydra:member'].flatMap((catalog) => {
      return catalog.localizedCatalogs.map((localizedCatalog) => {
        return {
          id: catalog.name,
          value: localizedCatalog['@id'],
          label: localizedCatalog.name,
        }
      })
    })

    const { container } = renderWithProviders(
      <ScopeSearchUsage catalogsOptions={catalogsOptions} />
    )

    expect(container).toMatchSnapshot()
  })
})
