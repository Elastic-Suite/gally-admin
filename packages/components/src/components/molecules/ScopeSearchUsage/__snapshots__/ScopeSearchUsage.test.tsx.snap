// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeSearchUsage match snapshot 1`] = `
<div>
  <div
    class="css-1rm3cd2"
    style="padding: 16px;"
  >
    <form
      class="css-1dl9duh"
      novalidate=""
    >
      <div
        class="MuiFormControl-root css-1girsym-MuiFormControl-root"
      >
        <div
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1qqsdnr-MuiAutocomplete-root"
        >
          <div
            class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-required css-rnc4nv-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r0:"
            >
              Scope
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ufxisk-MuiFormLabel-asterisk"
              >
                â€‰
                *
              </span>
            </label>
            <div
              class="InputText__Wrapper css-17fp4jk"
            >
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-jlza7d-MuiInputBase-root"
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-invalid="false"
                  autocapitalize="none"
                  autocomplete="off"
                  class="MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ngazp5-MuiInputBase-input"
                  id=":r0:"
                  placeholder="Select a localized catalog"
                  role="combobox"
                  spellcheck="false"
                  type="text"
                  value=""
                />
                <div
                  class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="form.open"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                    tabindex="-1"
                    title="form.open"
                    type="button"
                  >
                    <ion-icon
                      src="/images/ionicons/chevron-down.svg"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-vaijrj"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-s405n1-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1i75mai"
          >
             
            form.from
             
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
            >
              <div
                class="InputText__Wrapper css-17fp4jk"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-33c5sc-MuiInputBase-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInputBase-inputAdornedEnd css-ngazp5-MuiInputBase-input"
                    placeholder="date.placeholder"
                    required=""
                    type="tel"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        src="/images/ionicons/calendar-outline.svg"
                        style="font-size: 18px; padding: 0px;"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-ow4s8t"
          >
            form.to
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
            >
              <div
                class="InputText__Wrapper css-17fp4jk"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-33c5sc-MuiInputBase-root"
                  required=""
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInputBase-inputAdornedEnd css-ngazp5-MuiInputBase-input"
                    placeholder="date.placeholder"
                    required=""
                    type="tel"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        src="/images/ionicons/calendar-outline.svg"
                        style="font-size: 18px; padding: 0px;"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1jv49re-MuiButtonBase-root-MuiButton-root"
        data-testid="submitButton"
        style="margin-top: 8px;"
        tabindex="0"
        type="submit"
      >
        Apply
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
